tests/performance/density-kwok.go

```diff
@@ -0,0 +1,305 @@
+				corev1.ResourceMemory:           resource.MustParse("256Gi"),
+				corev1.ResourceEphemeralStorage: resource.MustParse("100Gi"),
+				corev1.ResourcePods:             resource.MustParse("110"),
+				"devices.kubevirt.io/kvm":       resource.MustParse("1k"),
+				"devices.kubevirt.io/tun":       resource.MustParse("1k"),
+				"devices.kubevirt.io/vhost-net": resource.MustParse("1k"),
+			},
+			Capacity: corev1.ResourceList{
+				corev1.ResourceCPU:              resource.MustParse("32"),
+				corev1.ResourceMemory:           resource.MustParse("256Gi"),
+				corev1.ResourceEphemeralStorage: resource.MustParse("100Gi"),
+				corev1.ResourcePods:             resource.MustParse("110"),
+				"devices.kubevirt.io/kvm":       resource.MustParse("1k"),
+				"devices.kubevirt.io/tun":       resource.MustParse("1k"),
+				"devices.kubevirt.io/vhost-net": resource.MustParse("1k"),
+			},
+		},
+	}
+
+	return node
+}
+
+func createFakeVMIBatchWithKWOK(kubevirtClient kubecli.KubevirtClient, vmCount int) {
+	for i := 1; i <= vmCount; i++ {
+		vmi := createFakeVMISpecWithResources()
+
+		_, err := kubevirtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(context.Background(), vmi, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		time.Sleep(100 * time.Millisecond)
+	}
+}
+
+func deleteAndVerifyFakeVMIBatch(kubevirtClient kubecli.KubevirtClient, vmCount int, timeout time.Duration) {
+	err := kubevirtClient.VirtualMachineInstance(util.NamespaceTestDefault).DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{})
+	if err != nil {
+		log.Fatal("Failed to delete VMIs")
+	}
+
+	Eventually(func() int {
+		vmis, err := kubevirtClient.VirtualMachineInstance(util.NamespaceTestDefault).List(context.Background(), metav1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		return len(vmis.Items)
+	}, timeout, 10*time.Second).Should(Equal(0))
+}
+
+func deleteAndVerifyFakeVMBatch(kubevirtClient kubecli.KubevirtClient, vmCount int, timeout time.Duration) {
+	err := kubevirtClient.VirtualMachine(util.NamespaceTestDefault).DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{})
+	if err != nil {
+		log.Fatal("Failed to delete VMs")
+	}
+
+	Eventually(func() int {
+		vmis, err := kubevirtClient.VirtualMachine(util.NamespaceTestDefault).List(context.Background(), metav1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		return len(vmis.Items)
+	}, timeout, 10*time.Second).Should(Equal(0))
+}
+
+func createFakeBatchRunningVMWithKwok(virtClient kubecli.KubevirtClient, vmCount int) {
+	for i := 1; i <= vmCount; i++ {
+		vmi := createFakeVMISpecWithResources()
+		vm := libvmi.NewVirtualMachine(vmi, libvmi.WithRunning())
+
+		_, err := virtClient.VirtualMachine(util.NamespaceTestDefault).Create(context.Background(), vm, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		// interval for throughput control
+		time.Sleep(100 * time.Millisecond)
```

Comment:
Will it otherwise overwhelm the system?

Reply:
Yes, since we are creating almost 1000 fake vmis, it is good to have sometime to process all the requests.